File: quide-specification/src/main/java/io/gitlab/arturbosch/quide/platform/Processor.java
package io.gitlab.arturbosch.quide.platform;

/**
 * @author Artur Bosch
 */
public interface Processor extends Executable, Nameable {

	default ControlFlow.InjectionPoint injectionPoint() {
		return ControlFlow.InjectionPoint.AfterAnalysis;
	}

	default int priority() {
		return 0;
	}
}

File: quide-api/src/main/kotlin/io.gitlab.arturbosch.quide.api/processors/Processor.kt
package io.gitlab.arturbosch.quide.api.processors

import io.gitlab.arturbosch.quide.api.Executable
import io.gitlab.arturbosch.quide.api.Nameable

/**
 * @author Artur Bosch
 */
interface Processor : Executable, Nameable {
	fun injectionPoint(): InjectionPoint = InjectionPoint.AfterAnalysis
	fun priority(): Int = 0
}
